lib/exq.ex

```diff
@@ -15,23 +15,8 @@ defmodule Exq do
 
   # Exq methods
   def start_link(opts \\ []) do
-    import Supervisor.Spec, warn: false
-
-    {redis_opts, connection_opts, opts} = conform_opts(opts)
-
-    # make sure redis always first(start in order)
-    children = [
-      worker(Redix, [redis_opts, connection_opts]),
-      worker(Exq.Middleware.Server, [opts]),
-      worker(Exq.Manager.Server, [opts]),
-      worker(Exq.Stats.Server, [opts]),
-      worker(Exq.Enqueuer.Server, [opts]),
-      supervisor(Exq.Worker.Supervisor, [opts])
-    ]
-
-    if opts[:scheduler_enable] do
-      children = children ++ [worker(Exq.Scheduler.Server, [opts])]
-    end
+    unless opts[:mode], do: opts = Keyword.put(opts, :mode, Config.get(:mode))
```

Comment:
I don't think we need this.  Can we do something like this in opts:
```
mode = opts[:mode] || Config.get(:mode)
```
And set the mode as :default on Config.ex?